nbs/animation.ipynb

```diff
@@ -215,34 +202,40 @@
         "[args_d.update(a.param.values()) for a in arg_objs]\n",
         "args=AnimationArgs(**args_d)\n",
         "\n",
-        "# to do: rename these where they're used in the animator\n",
-        "args.W=args.width\n",
-        "args.H=args.height\n",
-        "\n",
         "# create folder for frames output\n",
-        "timestring = datetime.datetime.now().strftime('%Y%m%d%H%M%S')\n",
-        "out_dir = os.path.join(outputs_path, timestring)\n",
-        "os.makedirs(out_dir, exist_ok=True)\n",
+        "if len(resume_timestring):\n",
+        "    out_dir = os.path.join(outputs_path, resume_timestring)\n",
+        "    if not os.path.exists(out_dir):\n",
+        "        raise Exception(\"Can't resume {resume_timestring} because path {out_dir} doesn't exist. Please make sure the timestring is correct.\")\n",
+        "    timestring = resume_timestring\n",
+        "else:\n",
+        "    timestring = datetime.datetime.now().strftime('%Y%m%d%H%M%S')\n",
+        "    out_dir = os.path.join(outputs_path, timestring)\n",
+        "    os.makedirs(out_dir, exist_ok=True)\n",
         "print(f\"Saving animation frames to {out_dir}...\")\n",
         "\n",
-        "artist = Animator(\n",
+        "animator = Animator(\n",
+        "    stub=stub,\n",
         "    animation_prompts=animation_prompts,\n",
         "    args=args,\n",
-        "    out_dir=out_dir,\n",
+        "    out_dir=out_dir,    \n",
         "    negative_prompt=negative_prompt,\n",
         "    negative_prompt_weight=negative_prompt_weight,\n",
+        "    resume=len(resume_timestring) != 0,\n",
         "    transform_engine_id=TRANSFORM_ENGINE_ID,\n",
         "    inpaint_engine_id=INPAINT_ENGINE_ID,\n",
         "    generate_engine_id=GENERATE_ENGINE_ID,\n",
-        "    )\n",
-        "artist.save_settings()\n",
-        "artist.setup_animation()\n",
-        "artist.render_animation(stub=stub)"
+        ")\n",
+        "animator.save_settings(f\"{timestring}_settings.txt\")\n",
```

Comment:
i think this will overwrite the previously written settings if we're resuming, right?

Reply:
correct. and that's a valid workflow, stopping animation, changing prompts or camera, then resuming and wanting latest in the settings.

Reply:
true, but it also collides with reproducibility, which I'd argue is something stored settings should at least very closely approximate. rather than overwriting settings entirely, we could change this behavior to identify what frame we're currently on and update the settings file to reflect that there was a change effective from that keyframe forward